src/components/PlayerListCreator.vue

```diff
@@ -0,0 +1,161 @@
+<template>
+  <div id="PlayerListCreator">
+    <div id="title">
+      <h1>
+        {{ $t("playerListCreator.title1_1") }}
+        <span>{{ $t("playerListCreator.title1_2") }}</span> ?
+      </h1>
+      <h2>
+        {{ $t("playerListCreator.title2") }}
+      </h2>
+    </div>
+    <div id="playerList">
+      <!-- Created player list -->
+      <draggable v-model="players">
+        <div
+          class="player"
+          v-for="(player, i) in players"
+          :key="i"
+          :id="'player_' + (i + 1)"
+        >
+          <div
+            :class="'name ' + (newPlayerName === player.getName() ? 'red' : '')"
+          >
+            {{ i + 1 + " : " + player.getName() }}
+          </div>
+          <div class="removeBtn">
+            <md-button
+              v-on:click="removePlayer(i)"
+              class="removePlayerBtn md-icon-button"
+            >
+              <md-icon>remove</md-icon>
+            </md-button>
+          </div>
+        </div>
+      </draggable>
+      <!-- Add player line -->
+      <form class="player add" @submit.prevent="">
+        <md-field class="name">
+          <md-input type="text" name="newPlayerName" v-model="newPlayerName"></md-input>
+        </md-field>
+        <div class="addBtn">
+          <md-button
+            type="submit"
+            id="addPlayerBtn"
+            @click="addPlayer"
+            class="md-icon-button"
+          >
+            <md-icon>add</md-icon>
+          </md-button>
+        </div>
+      </form>
+    </div>
+  </div>
+</template>
+
+<script>
+import draggable from 'vuedraggable'
+import Player from '../objets/player'
+
+export default {
+  name: 'PlayerListCreator',
+  components: {
+    draggable
+  },
+  data: () => ({
+    newPlayerName: '',
+    players: []
+  }),
+  methods: {
+    addPlayer () {
+      // Add a new player object to the player list
+      if (
+        this.newPlayerName.length > 0 &&
+        this.players.find((p) => p.getName() === this.newPlayerName) ===
+          undefined
+      ) {
+        // check for duplicates
+        this.players.push(new Player(this.newPlayerName))
+        this.newPlayerName = ''
+      }
+    },
+    removePlayer (index) {
+      // Remove a player from his index pos
+      if (index >= 0 && index < this.players.length) {
```

Comment:
L'indice ici peut être négatif ? La deuxième condition suffit non ?

Reply:
dans le doute, le but c'est d’empêcher le out of bound